### PR TITLE
Feat extend set extra info

### PR DESCRIPTION
**Pull request by Samuel-Oglegba**
This PR seeks to extend set_extra_info as follows:

- add an optional input parameter set_new_var
- when set_new_var is TRUE, and the required input `_variable_` is not in the symbol table
- call set_new_info to add the variable to the symbol table

### CODE REVIEW

File: lib/analysis.pi [hunk 1]
```diff
@@ -16,7 +16,9 @@
 
 <xform get_extra_info pars=(variable, tag) tab=GLOBAL.SymTable/>
 
-<xform set_extra_info pars=(variable, info, tag) tab=GLOBAL.SymTable/>
+<xform set_extra_info pars=(variable, info, tag) set_new_var=FALSE tab=GLOBAL.SymTable/>
```
Comment (qingyi-yan) [inline lib/analysis.pi:19]: Change set_new_var to insert_new_var

File: lib/analysis.pt [hunk 2]
```diff
@@ -193,8 +193,12 @@ switch (vars)
   }
 </xform>
 
-<xform set_extra_info pars=(variable, info, tag) tab=GLOBAL.SymTable>
+<xform set_extra_info pars=(variable, info, tag) set_new_var=FALSE tab=GLOBAL.SymTable>
   scope = get_scope(variable);
+  if(scope == "" && (set_new_var:TRUE)){
```
Comment (qingyi-yan) [inline lib/analysis.pt:198]: Change 
(set_new_var:TRUE)

to insert_new_var

**Review by qingyi-yan — APPROVED** [review]

File: lib/analysis.pi (hunk 1 above)
Comment (Samuel-Oglegba) [reply lib/analysis.pi]: variable name was changed

File: lib/analysis.pt (hunk 2 above)
Comment (Samuel-Oglegba) [reply lib/analysis.pt]: The variable name was changed to insert_new_var.

**Pull request merged**